src/controllers/session.ts

```diff
@@ -112,67 +116,95 @@ const deleteSession = async (req: Request, res: Response) => {
   );
-  if (!sessionUser) {
-    const newSessionUser = await SessionModel.findOneAndUpdate(
-      { _id: sessionId },
-      {
-        $addToSet: {
-          participant: {
-            user: {
-              userInfo: req.user.userId,
-              userStatus,
-            },
-          },
-        },
-      },
-      { new: true }
+
+  let event;
+  if (status === true) {
+    if (!participantIds.includes(userId)) {
+      (session.participant as any).push(user._id);
+      await session.save();
+
+      // const populateCreatorOptions = {
+      //   path: 'creator',
+      //   select: 'name ',
+      // };
+
+      // const sessionInfo = (await SessionModel.findById({
+      //   _id: sessionId,
+      // }).populate(populateCreatorOptions)) as IPopulatedSession;
+
+      // const sessionStart = sessionInfo?.start;
+      // const sessionEnd = sessionInfo?.end;
+      // const sessionType = sessionInfo?.type;
+      // const sessionCreator = sessionInfo?.creator.name;
+
+      // if (!sessionType || !sessionEnd || !sessionStart) {
+      //   throw new BadRequestError('Session does not exist');
+      // }
+
+      // event = await scheduleEvent({
+      //   summary: sessionType,
+      //   start: sessionStart,
+      //   end: sessionEnd,
+      //   email: req.user.email,
+      //   description: sessionCreator,
+      // });
+    }
+  } else if (status === false) {
+    const userIndex = session.participant.findIndex((participant) =>
+      (participant as any).equals(userId)
     );
-    if (!newSessionUser) {
-      throw new BadRequestError('Can not update this session');
+    if (userIndex !== -1) {
+      session.participant.splice(userIndex, 1);
+      await session.save();
     }
+  } else {
+    return res.status(400).json({ message: 'Invalid status value' });
+  }
 
-    let event;
-    if (sessionUser || newSessionUser) {
-      if (userStatus !== 'confirm') {
-        throw new BadRequestError('Can not update your calendar');
-      }
-      const populateCreatorOptions = {
-        path: 'creator',
-        select: 'name ',
-      };
-
-      const sessionInfo = (await SessionModel.findOne({
-        _id: sessionId,
-      }).populate(populateCreatorOptions)) as IPopulatedSession;
-
-      const sessionStart = sessionInfo?.start;
-      const sessionEnd = sessionInfo?.end;
-      const sessionType = sessionInfo?.type;
-      const sessionCreator = sessionInfo?.creator.name;
-
-      if (!sessionType || !sessionEnd || !sessionStart) {
-        throw new BadRequestError('Session does not exist');
-      }
-      event = await scheduleEvent({
-        summary: sessionType,
-        start: sessionStart,
-        end: sessionEnd,
-        email: req.user.email,
-        description: sessionCreator,
-      });
-    }
+  return res.status(200).json({ session, event });
+};
 
-    return res.status(201).json({ newSessionUser, event });
+const getStatus = async (req: Request, res: Response) => {
```

Comment:
This method is fine, but we could also check the user status in the Front-end. In there, we can directly check the currently logged userId against the list of userIds of each session.